src/main/groovy/life/qbic/utils/NanoporeParser.groovy

```diff
@@ -0,0 +1,283 @@
+package life.qbic.utils
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import groovy.json.JsonSlurper
+import groovy.util.logging.Log4j2
+import groovyjarjarcommonscli.MissingArgumentException
+import org.everit.json.schema.Schema
+import org.everit.json.schema.ValidationException
+import org.everit.json.schema.loader.SchemaLoader
+import org.json.JSONObject
+import org.json.JSONTokener
+
+import java.nio.file.NotDirectoryException
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.text.ParseException
+import life.qbic.datamodel.datasets.OxfordNanoporeExperiment
+
+@Log4j2
+class NanoporeParser {
+
+    final static private JSON_SCHEMA = "/nanopore-instrument-output.schema.json"
+
+    /**
+     * Generates a map representing the folder structure
+     * @param directory path of directory whose fileTree should be converted into map
+     */
+    static OxfordNanoporeExperiment parseFileStructure(Path directory) {
+        // Step1: convert directory to json
+        Map convertedDirectory = DirectoryConverter.fileTreeToMap(directory)
+
+        String json = mapToJson(convertedDirectory)
+        try {
+            validateJsonForSchema(json, JSON_SCHEMA)
+            //Step3: convert valid json to OxfordNanoporeExperiment Object
+
+            // Step4: Parse meta data out of report files and extend the map
+            def finalMap = parseMetaData(convertedDirectory, directory)
+
+            // Step5: Create the final OxfordNanoporeExperiment from the map
+            OxfordNanoporeExperiment convertedExperiment = OxfordNanoporeExperiment.create(finalMap)
+            return convertedExperiment
+        } catch (ValidationException validationException) {
+            log.error("Specified directory could not be validated")
+            // we have to fetch all validation exceptions
+            def causes = validationException.getCausingExceptions().collect{ it.message }.join("\n")
+            log.error(causes)
+            throw validationException
+        }
+    }
+
+
```

Comment:
Should we provide @param and @return documentation for all metadata handling methods?

Reply:
yes!!